### PR TITLE
MDEV-12386: Use AppVeyor for public Windows CI

### PR DESCRIPTION
**Pull request by grooverdan**
This is so pull requests will be Windows tested.

Current Build: https://ci.appveyor.com/project/grooverdan/mariadb-server

Obviously needs a few clicks to authorise AppVeyor as a github hook.

In the AppVeyor the settings I changed where:

"Skip branches without appveyor.yml" - enable  (not sure how it would work without this).
"Git clone depth" - 2 (though seems to be ignored looking at output)

I submit this under the MCA.

### CODE REVIEW

**Comment by svoj** [comment]
I believe only @vuvova has access to our repo configuration.
@grooverdan it would be nice if you could elaborate what exactly has to be done on github side.

**Comment by grooverdan** [comment]
Using https://ci.appveyor.com/, github login and it will request use github's OAuth to request permissions for the projects. After that appveyor will have settings for MariaDB/server. Defaults are pretty good however recommend changing the above settings as it seems more robusts/efficient.

**Comment by howard0su** [comment]
Do we have any update on this?

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+before_build:
+- md %APPVEYOR_BUILD_FOLDER%\win_build
+- cd %APPVEYOR_BUILD_FOLDER%\win_build
+- cmake ..  -G "Visual Studio 15 2017" -DBOOST_ROOT=C:\Libraries\boost_1_64_0 -DPKG_CONFIG_EXECUTABLE=C:\perl\bin\pkg-config -DBISON_EXECUTABLE=C:\cygwin\bin\bison
+- dir
+build:
+  project: win_build\MySQL.sln
+  parallel: true
+  verbosity: minimal
+test:
```
Comment (svoj) [inline appveyor.yml]: configuration: Debug

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+version: 10.2.{build}
```
Comment (svoj) [inline appveyor.yml]: version: 10.2-{build} or 10.2:build or whatever

File: appveyor.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+image: Visual Studio 2017
+before_build:
```
Comment (svoj) [inline appveyor.yml]: move image: field down, e.g. to the very end

File: appveyor.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+version: 10.2-{build}
+before_build:
+- md %APPVEYOR_BUILD_FOLDER%\win_build
+- cd %APPVEYOR_BUILD_FOLDER%\win_build
+- cmake ..  -G "Visual Studio 15 2017" -DBOOST_ROOT=C:\Libraries\boost_1_64_0 -DPKG_CONFIG_EXECUTABLE=C:\perl\bin\pkg-config -DBISON_EXECUTABLE=C:\cygwin\bin\bison
```
Comment (vaintroub) [inline appveyor.yml]: PKG_CONFIG does not really work on Windows, Strawberry perl's uses mingw  libraries, which VS  compiler cannot use,
so PKG_CONFIG_EXECUTABLE is probably best NOT-FOUND

Comment (vaintroub) [reply appveyor.yml]: Also, BOOST_ROOT is not really needed. Why it is there?

Comment (grooverdan) [reply appveyor.yml]: Thanks @vaintroub. As usual your expertise is invaluable.

**Comment by grooverdan** [comment]
Success: https://ci.appveyor.com/project/grooverdan/mariadb-server/build/10.2-12

**Comment by vuvova** [comment]
merged in bb-10.2-serg, will be in 10.2 later today

**Pull request closed without merging**